### PR TITLE
Fix push_nested_display_list's glyph cache handling (and also the glyph cache parser)

### PR DESCRIPTION
**Pull request by Gankro**
* Cache stores indices, not full GlyphInstances
* push_nested needs to split off the cache and data and handle them seperately

### CODE REVIEW

**Comment by Gankro** [comment]
Post-mortem: 

* nothing in our codebase tests push_nested_display_list
* nothing in our codebase currently parses the cache (should be fixed when @glennw tries to use it)

File: webrender_api/src/display_list.rs [hunk 1]
```diff
@@ -289,15 +297,16 @@ impl<'a> BuiltDisplayListIter<'a> {
 }
 
 impl<'a> Iterator for GlyphsIter<'a> {
-    type Item = (FontKey, ColorF, ItemRange<GlyphInstance>);
+    // NOTE: u32 = GlyphInstance.index
```
Comment (kvark) [inline webrender_api/src/display_list.rs]: let's have a typedef?

File: webrender_api/src/display_list.rs [hunk 2]
```diff
@@ -289,15 +297,16 @@ impl<'a> BuiltDisplayListIter<'a> {
...
+    type Item = (FontKey, ColorF, ItemRange<u32>);
```
Comment (kvark) [inline webrender_api/src/display_list.rs]: wouldn't it be simpler to have `AuxIter<>` as the last element in this tuple?

**Comment by glennw** [comment]
@Gankro I think bors got confused by the travis mac issues earlier today. I usually do a ```git commit --amend``` and ```git push -f``` to re-trigger the CI. There might be a simpler way, but that usually does the trick.

**Pull request merged**